compiler/src/dotty/tools/backend/jvm/BCodeBodyBuilder.scala

```diff
@@ -755,23 +770,23 @@ trait BCodeBodyBuilder extends BCodeSkelBuilder {
         case app @ Apply(fun, args) =>
           val sym = fun.symbol
 
-          if (sym.isLabel) { // jump to a label
-            assert(int.hasLabelDefs)
-            genLoadLabelArguments(args, labelDef(sym), app.pos)
-            bc goTo programPoint(sym)
-          } else if (isPrimitive(fun)) { // primitive method call
+          if (isPrimitive(fun)) { // primitive method call
             generatedType = genPrimitiveOp(app, expectedType)
           } else { // normal method call
             val invokeStyle =
               if (sym.isStaticMember) InvokeStyle.Static
-              else if (sym.isPrivate || sym.isClassConstructor) InvokeStyle.Special
+              else if (sym.is(Flags.Private) || sym.isClassConstructor) InvokeStyle.Special
               else InvokeStyle.Virtual
 
             if (invokeStyle.hasInstance) genLoadQualifier(fun)
             genLoadArguments(args, paramTKs(app))
 
-            val Select(qual, _) = fun // fun is a Select, also checked in genLoadQualifier
-            if (isArrayClone(fun)) {
+            val DesugaredSelect(qual, _) = fun // fun is a Select, also checked in genLoadQualifier
+            val isArrayClone = fun match {
+              case DesugaredSelect(qual, nme.clone_) if qual.tpe.widen.isInstanceOf[JavaArrayType] => true
```

Comment:
DesugaredSelect used on the same tree twice.

Reply:
Removed

Reply:
Removed